### PR TITLE
print stats

### PR DESCRIPTION
**Pull request by knguyen142**
Prints out the statistics of the run at the end or on exit. 

Sample:

```
^CShutting Down...
======
attempted count:5
success count: 5
error count: 0
======
ButtonMissingError count: 0
TimeoutError count: 0
Errors not defined: 0
```

Sam, the key difference we made here from what I showed you is we separated the code so if we do a ctl-C (exit the program early) we can also get the stats so far as opposed to having to wait until the end. Will wait for you to test it before merging.

### CODE REVIEW

File: vci-puppeteer/vci-approve-variants.js [hunk 1]
```diff
@@ -28,6 +29,33 @@ const DOMAIN_CONFIG = {
...
 
+class ButtonMissingError extends Error {
+  constructor(message) {
+    super(message); // (1)
+    this.name = "ButtonMissingError"; // (2)
+  }
+}
+
+// Keept track of stats on the run to print out at the end or on exit.
+let runStatistics = {
```
Comment (knguyen142) [inline vci-puppeteer/vci-approve-variants.js:40]: Sam, we made this global now (outside of a function, before it was in the function making the var local to that function). Now we have access to it wherever, namely at the block where we intercept an exit ctl-C command  (defined in `process.on( 'SIGINT',`).

File: vci-puppeteer/vci-approve-variants.js [hunk 2]
```diff
@@ -307,6 +342,30 @@ function main() {
 	})();
 }
 
+function printRunStatistics() {
```
Comment (knguyen142) [inline vci-puppeteer/vci-approve-variants.js:345]: This function is called after we iterate through every variant (so at the end of the variants list) or if the user is terminating the program.

File: vci-puppeteer/vci-approve-variants.js [hunk 3]
```diff
@@ -307,6 +342,30 @@ function main() {
...
+		if (runStatistics.errors[subError].type !== Error) {
+			console.log(subError + ' count: ' + runStatistics.errors[subError].count);
+			definedErrors += runStatistics.errors[subError].count;
+		}
+	}
+	console.log('Errors not defined: ' + (errorCount - definedErrors));
+}
+
+// If exiting, print out run statistics.
+process.on( 'SIGINT', function() {
```
Comment (knguyen142) [inline vci-puppeteer/vci-approve-variants.js:364]: This is what is called when the user tries to exit the program (ctl-c).

**Review by larrybabb — APPROVED** [review]
I didn't run the code, I'll leave that to Sam. All looks good, nice job.

**Pull request merged**